transformer_engine/common/fused_attn/fused_attn.cpp

```diff
@@ -34,16 +36,18 @@ void nvte_fused_attn_fwd_qkvpacked(
   Tensor *output_O = reinterpret_cast<Tensor*>(O);
   Tensor *wkspace = reinterpret_cast<Tensor*>(workspace);
 
-  // QKV shape is [total_seqs, 3, h, d]
+  // QKV shape is either [total_seqs, 3, h, d] or [b, s, 3, h, d]
```

Comment:
Since we are getting `b` from elsewhere, can we stay with `total_seqs` here (as in, `b*s` in this case)?

Reply:
The current logics for extracting the dimensions is [here](https://github.com/NVIDIA/TransformerEngine/blob/d717f535ea0c9c50ef426c6419182791ea9e9092/transformer_engine/common/fused_attn/fused_attn.cpp#L40-L43)
```
  auto ndim = input_QKV->data.shape.size();
  size_t b = input_cu_seqlens->data.shape[0] - 1;
  size_t h = input_QKV->data.shape[ndim - 2];
  size_t d = input_QKV->data.shape[ndim - 1];
```
It seems this rule can handle the memory layout of [total_seqs, 3, h, d], [b*s, 3, h, d], [s*b, 3, h, d] and even [b, s, 3, h, d] or [s, b, 3, h, d].

Reply:
Nitpick: we still need to remove the "either"s from the comments

Reply:
Removed in b13be070cbc1c32921321497f07a7c76e1cb5dee